bin/add_frontend_server.rb

```diff
@@ -0,0 +1,120 @@
+require 'aws-sdk'
+require_relative '../deployment'
+
+MAX_WAIT_TIME = 600 #Wait no longer than 10 minutes for instance creation. Typically this takes a couple minutes
+
+Aws.config.update({
+                      region: CDO.aws_region,
+                      credentials: Aws::Credentials.new(CDO.aws_access_key, CDO.aws_secret_key),
+                  })
+
+ec2client = Aws::EC2::Client.new
+
+instances = ec2client.describe_instances
+
+#Determine distribution of availability zones, pick the one that has the least capacity among frontend instances
+instance_distribution = Hash.new(0)
+
+instances.reservations.each do |reservation|
+  reservation.instances.each do |instance|
+    unless instance.state.name == 'running'
+      next
+    end
+
+    is_frontend = true
+
+    instance.tags.each do |tag|
+      if tag.key == 'Name' and not tag.value.include? 'frontend'
+        is_frontend = false
+        break
+      end
+    end
+
+    unless is_frontend
+      next
+    end
+
+    availability_zone = instance.placement.availability_zone
+    if instance_distribution.has_key?(availability_zone)
+      instance_distribution[availability_zone] += 1
+    else
+      instance_distribution[availability_zone] = 1
+    end
+  end
+end
+
+puts "Current availability zone distribution #{instance_distribution}"
+
+determined_instance_zone = instance_distribution.select { |_, v| v == instance_distribution.values.min }.keys[0]
+
+puts "Using underscaled instance zone #{determined_instance_zone}"
```

Comment:
I don't think that'll work - there are potentially multiple reservations per instance